stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+from awacs.aws import (
+    Allow,
+    Statement,
+    Principal,
+    Policy
+)
+from awacs.sts import (
+    AssumeRole
+)
+
+from troposphere import (
+    Base64,
+    FindInMap,
+    GetAtt,
+    Join,
+    Ref,
+    Output
+)
+
+from troposphere.autoscaling import (
+    AutoScalingGroup,
+    LaunchConfiguration,
+    Tag
+)
+
+from troposphere.ec2 import (
+    BlockDeviceMapping,
+    EBSBlockDevice
+)
+
+from troposphere.iam import (
+    InstanceProfile,
+    Role
+)
+
+from troposphere.policies import (
+    AutoScalingReplacingUpdate,
+    AutoScalingRollingUpdate,
+    UpdatePolicy
+)
+
+from troposphere import eks
+from stacker.blueprints.base import Blueprint
+
+
+class Cluster(Blueprint):
+    VARIABLES = {
+        "Name": {
+            "type": str,
+            "description": "The name of the cluster to create.",
+        },
+        "ExistingRoleArn": {
+            "type": str,
+            "description": "IAM Role ARN with EKS assume role policies. One "
+                           "will be created if it's not provided.",
+            "default": ""
+        },
+        "Version": {
+            "type": str,
+            "description": "Kubernetes version",
+            "default": "",
+        },
+        "SecurityGroupIds": {
+            "type": str,
+            "description": "A comma separated list of security group ids for "
+                           "controlling ENI access from EKS to workers",
+        },
+        "SubnetIds": {
+            "type": str,
+            "description": "A comma separated list of subnet ids where you "
+                           "will launch your worker nodes",
+        }
+    }
+
+    # This creates an IAM role which EKS requires, as described here:
+    # https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html#eks-create-cluster
+    def create_iam_role(self):
+        eks_service_role_id = "EksServiceRole"
+        t = self.template
+        t.add_resource(
```

Comment:
A newer pattern in troposphere that we're making more use of in stacker is to assign the creation of resources to python variables, then use the builtin troposphere `Ref` and `GetAtt` methods.  An example for this:

```
role = t.add_resource(Role(...))
return role.GetAtt("Arn")
```

Reply:
done